### PR TITLE
Support Hasura v2.2 action JSON `request_query` field

### PR DESCRIPTION
**Pull request by pcrosemurgy**
* **Tickets addressed:** N/A
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
Supports new Hasura action handler fields (mainly `request_query`): https://hasura.io/docs/latest/graphql/core/actions/action-handlers.html#http-handler

## Verification
Simulation run through UI runs without issue. This serves as a decent end-to-end test for this PR.

## Documentation
None.

## Future work
None.

### CODE REVIEW

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java
Comment (mattdailis) [inline merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java]: 🔶 A "discussion question" of sorts is - is `requestQuery` truly optional, or does `v2.2` always provide it?

The optional seems like a reasonable way to support both `v2.0.10` and `v2.2`, but I just want to make sure I understand the motivation.

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java
Comment (mattdailis) [inline merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java:140]: 🔶 Would it make any sense to add the `requestQuery` as a field on `HasuraAction`? Or do we prefer to discard it, since we have no use for it?

**Review by mattdailis — APPROVED** [review]
Looks good to me.

Legend:
🔻 Issues to address before merge
🔶 Requests that should not block merge, but should at least be discussed
🔵 Recommendations that can be ignored if desired

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java
Comment (pcrosemurgy) [reply merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java:140]: At first that's what I did, but until we have a use for it I think it's probably best to discard it.

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java
Comment (pcrosemurgy) [reply merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java]: Yeah it's nice that this would allow us to downgrade to v2.0 if we ever needed to. Am open to making this a plain old `field()` and discarding it instead. No strong opinion here so lmk if you have any preference :)

**Review by camargo — APPROVED** [review]
Tested simulation action and it works as expected. LGTM.

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java
Comment (mattdailis) [reply merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java]: 🔻 I think it's fine, but write a comment explaining the reasoning, and I think we'll want to eventually pick one version to support.

**Comment by goetzrrGit** [comment]
I seem to have broken something on my end so I couldn't test it. No activity_types show up in Hasura and I can't add anything to a plan. No need to block merging though.

**Comment by pcrosemurgy** [comment]
@goetzrrGit thanks for testing this out, I'll double check and make sure I don't see the same issue. You may need to remove all your Docker images, volumes, etc. (since we've upgraded the Hasura image).

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java
Comment (pcrosemurgy) [reply merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java]: Just pushed a change to make this a `field()` :)

**Comment by goetzrrGit** [comment]
@pcrosemurgy I nuked everything. Here is how I ran it on my computer.

`docker system prune --all --volumes   `
`./gradlew build; docker compose -f docker-compose.yml build; docker compose -f docker-compose.yml up
`

**Comment by pcrosemurgy** [comment]
Hmm, what's your output when running `docker images`?
I'm thinking you may still have the old Hasura image. My output looks like:
```
REPOSITORY                    TAG                        IMAGE ID       CREATED         SIZE
aerie_merlin                  latest                     c02801e69e00   3 seconds ago   578MB
ghcr.io/nasa-ammos/aerie-ui   develop                    ba1ec4c473b6   2 hours ago     363MB
hasura/graphql-engine         v2.2.0.cli-migrations-v3   ebdf522b135f   8 days ago      462MB
```

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java
Comment (pcrosemurgy) [reply merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java]: Also retested with this change. Looking good

**Comment by goetzrrGit** [comment]
I think I have the latest:

```
REPOSITORY                         TAG                        IMAGE ID       CREATED          SIZE
aerie_merlin                       latest                     c37b031f2657   25 minutes ago   578MB
aerie_scheduler                    latest                     03c139cbeaeb   25 minutes ago   608MB
ghcr.io/nasa-ammos/aerie-ui        develop                    ba1ec4c473b6   2 hours ago      363MB
ghcr.io/nasa-ammos/aerie-gateway   develop                    c36efce8e44f   4 hours ago      147MB
hasura/graphql-engine              v2.2.0.cli-migrations-v3   ebdf522b135f   8 days ago       462MB
postgres                           14.1                       da2cb49d7a8d   2 weeks ago      374MB
```

**Comment by pcrosemurgy** [comment]
Sorry Ryan, really not too sure here. To be honest, when spinning up a local Docker stack I completely remove all images, volumes, containers, networks, etc. It takes ~3 minutes but while waiting I can generally work on other things. I am wondering if your `aerie_merlin` image is too old? It looks like it was created 25 mins. ago (from when you ran `docker images`) but I think you tested things out ~10 mins. prior to checking `docker images`. @camargo is my goto for Docker stack issues, he may be able to spot the issue.

**Edit** assuming that the `aerie_merlin` image was created after checking out this branch and running `gradle clean build` then my comment about its `CREATED` timestamp is probably irrelevant.

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java
Comment (mattdailis) [reply merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinParsers.java]: Looks fine to me - as long as we don't need to support older versions of hasura (I'm not the right person to make that decision)

**Review by mattdailis — APPROVED** [review]
Giving a second approval, since I ended up requesting changes. 

I think it's okay to break support for older versions of hasura. We're still an alpha project, and don't want to bloat our code with too much baggage just yet.

**Pull request merged**
